test/perf/README.adoc

```diff
@@ -0,0 +1,66 @@
+
+== How performance is measured
+
+All metrics collected during the performance tests come from the `/metrics` endpoint provided by the metrics server running within the operator pod.
+Each controller automatically exposes its own metrics via the Operator SDK, so we don't need to add anything in our code, unless we want to expose extra data.
+
+While the metrics evolve with time, the `/metrics` endpoint only provides "instantaneous" values so we need to poll the data from this endpoint to track changes (eg: memory or CPU) or wait until a certain value or threshold is reached (eg: number of resources processed).
+
+A `Service` is associated with the `/metrics` endpoint on each operator pod, and the performance test creates a `Route` to expose the service. This is a requirement since the test runs outside of the cluster.
+
+Here's an example of the metrics for the Host operator:
+
+----
+...
+
+# HELP process_resident_memory_bytes Resident memory size in bytes.
+# TYPE process_resident_memory_bytes gauge
+process_resident_memory_bytes 6.1227008e+07
+# HELP process_virtual_memory_bytes Virtual memory size in bytes.
+# TYPE process_virtual_memory_bytes gauge
+process_virtual_memory_bytes 7.62011648e+08
+# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
+# TYPE process_virtual_memory_max_bytes gauge
+process_virtual_memory_max_bytes -1
+
+...
+
+# HELP controller_runtime_reconcile_total Total number of reconciliations per controller
+# TYPE controller_runtime_reconcile_total counter
+controller_runtime_reconcile_total{controller="masteruserrecord-controller",result="success"} 24
+controller_runtime_reconcile_total{controller="notification-controller",result="success"} 18
+controller_runtime_reconcile_total{controller="nstemplatetier-controller",result="success"} 3
+controller_runtime_reconcile_total{controller="registrationservice-controller",result="success"} 2
+controller_runtime_reconcile_total{controller="templateupdaterequest-controller",result="success"} 30
+controller_runtime_reconcile_total{controller="toolchaincluster-controller",result="success"} 16
+controller_runtime_reconcile_total{controller="toolchainstatus-controller",result="requeue_after"} 30
+controller_runtime_reconcile_total{controller="usersignup-controller",result="success"} 40
+
+...
+
+----
+
+== What is measured
+
+=== Processing existing resources after a new deployment
+
+To measure how much time a controller takes to process existing resources after a new deployment, the test does the following steps:
+
+1. Provision the user accounts.
+2. Create (if needed) a route to expose the `/metrics` endpoint to the outside world.
+3. Delete the operator pod. The Deployment controller will immediately trigger a new pod.
+4. Wait until the `/metrics` endpoint is available. At this point, the controller has fully bootstrapped, so the test can start the stopwatch.
```

Comment:
I guess this step is no longer accurate since you moved the `hostAwait.SetupRouteForService(metricsService, "/metrics")` before deleting the operator pod?

Reply:
no, we want to wait until the endpoint is available *again*, ie, when the pod has finished bootstrapping after the deletion in step 3. 
Maybe this sentence should be reworded to clarify?

Reply:
I think the sentence is correct, I just don't think the code is reflecting that anymore since you moved the `hostAwait.SetupRouteForService(metricsService, "/metrics")` up? 

Before this PR we captured the start time `start := time.Now()` after the metrics endpoint is available (after the operator pod was deleted and comes back up) but with the changes in this PR `start := time.Now()` would be called right after we delete the operator pod so the metrics endpoint may not be available when we capture the start time?

Reply:
@rajivnathan sorry for the late response, this PR totally slipped out of my mind. I reworded the steps in fabffcc. Please let me know if it's ok for you like that now.

Reply:
Yeah, that sounds accurate now. Thanks! One minor thing though is it incorrectly says the member operator pod when it's actually the host operator pod that is deleted.